components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/extension/eidas/EidasExtensionProcessor.java

```diff
@@ -0,0 +1,354 @@
+    private static String errorMsg = "Mandatory Attribute not found.";
+
+    /**
+     * Check whether the SAML authentication request can be handled by the eIDAS extension processor.
+     *
+     * @param request SAML request
+     * @return true if the request can be handled
+     * @throws IdentitySAML2SSOException
+     */
+    @Override
+    public boolean canHandle(RequestAbstractType request) throws IdentitySAML2SSOException {
+
+        return request.getNamespaces().stream().anyMatch(namespace -> namespace.getNamespaceURI()
+                .equals(EidasConstants.EIDAS_NS));
+    }
+
+    /**
+     * Check whether the SAML response can be handled by the eIDAS extension processor.
+     *
+     * @param authReqDTO Authentication request data object
+     * @return true if the request can be handled
+     * @throws IdentitySAML2SSOException
+     */
+    @Override
+    public boolean canHandle(StatusResponseType response, Assertion assertion, SAMLSSOAuthnReqDTO authReqDTO)
+            throws IdentitySAML2SSOException {
+
+        return authReqDTO.getProperties().getProperty(EidasConstants.EIDAS_REQUEST)
+                .equals(EidasConstants.EIDAS_PREFIX);
+    }
+
+    /**
+     * Process and Validate the SAML extensions in authentication request for EIDAS message format.
+     *
+     * @param request        Authentication request
+     * @param validationResp reqValidationResponseDTO
+     * @throws IdentityException
+     */
+    @Override
+    public void processSAMLExtensions(RequestAbstractType request, SAMLSSOReqValidationResponseDTO validationResp)
+            throws IdentitySAML2SSOException {
+
+        if (request instanceof AuthnRequest) {
+            if (log.isDebugEnabled()) {
+                log.debug("Process the extensions for EIDAS message format");
+            }
+            Extensions extensions = request.getExtensions();
+            if (extensions != null) {
+                validateForceAuthn(validationResp);
+                validateIsPassive(validationResp);
+                validateAuthnContextComparison(validationResp);
+                validateSPType(validationResp, extensions);
+
+                processRequestedAttributes(validationResp, extensions);
+            }
+        }
+    }
+
+    /**
+     * Process and Validate a response against the SAML request with extensions for EIDAS message format.
+     *
+     * @param response    SAML response
+     * @param assertion   SAML assertion
+     * @param authReqDTO Authentication request data object
+     * @throws IdentitySAML2SSOException
+     */
+    @Override
+    public void processSAMLExtensions(StatusResponseType response, Assertion assertion, SAMLSSOAuthnReqDTO authReqDTO)
+            throws IdentitySAML2SSOException {
+
+        if (response instanceof Response) {
+            if (log.isDebugEnabled()) {
+                log.debug("Process the extensions for EIDAS message format");
```

Comment:
This does not indicate if it's in the response flow. If SP information  is available here mention them in debug logs